### ISSUE TITLE
Simulation executes 2x as many times as expected

### ISSUE THREAD

**Issue by antoniohramirez** [gmucsn/mTree#33]
It looks like basic_simulation.json runs twice as many times as we expected.

We have "number_of_runs": 2 in the config file and have the mTree runner set to -m, but we have 4 outputs in the log file.

Why does this happen?

**Comment by skunath** [comment]
Can you send the relevant log output?

I'd check to see if you have any other references to the experiment log to
make sure there's no additional outputs.

**Comment by antoniohramirez** [comment]
[mtree.log](https://github.com/gmucsn/mTree/files/3423772/mtree.log)

**Comment by skunath** [comment]
Which branch is this code on?

**Comment by antoniohramirez** [comment]
We just pushed to master after making some structural changes to our agent strategy classes. It looks like we don't have the running 2x problem anymore, but now it only runs 1-2 more times than we expect. In simulation_launch.json, we expect 12 runs, but the mtree.log gives us 13-14 outputs

**Comment by skunath** [comment]
OK. So you have like a baker's dozen sims? Isn't that a better value?

I'll pull the code when I get home and see if there is any problem with the
sim dispatcher.

**Comment by antoniohramirez** [comment]
Did you find anything in that sim dispatcher?

**Comment by skunath** [comment]
So so far I've not been able to replicate your numbers. In the current
config I only get three outputs which would seem to be as expected.

I've almost got the simulation_id and run_number problem fixed. Hopefully I
can get a patch before noon.

**Comment by apsurek** [comment]
We have tested that this issue (running multiple times) persists across both Mac and Windows. Maybe try running it with a higher number of runs and running multiple configs. We have created a config file called "erroring_setup.json" which should help you replicate the error. All of this code is on the master branch, so if you pull from there and run the config, you should see this error.

**Comment by skunath** [comment]
I'm still getting the right number -- 400.

Let me get the simulation_id and run_number id and maybe that could help
debug yours.

**Comment by skunath** [comment]
Also, if you pull the latest updates to mTree you should get some new
properties on the institution.

These two should probably be included in your log statements now. Hopefully
that can help us track down the extra runs issue.

self.simulation_id
self.run_number

**Comment by apsurek** [comment]
I just pulled mTree 3 seconds ago and there was no updated to pull.

**Comment by skunath** [comment]
Sorry. Forgot to do the pull request. Try again now.

**Comment by antoniohramirez** [comment]
Try erroring_setup_2.json. I just pushed it to master, and it ran 280 times when I specified 240

**Comment by skunath** [comment]
Will do. 

Were you able to add the run number and simulation id to the log?

**Comment by antoniohramirez** [comment]
Yes, the run_number and simualtion_id work -- it helped us debug the issue
